quic/s2n-quic-sim/src/main.rs

```diff
@@ -1,17 +1,26 @@
 // Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 // SPDX-License-Identifier: Apache-2.0
 
+#[cfg(feature = "unstable-provider-io-testing")]
```

Comment:
All of this feature gating is excessive. There has to be a better way. For example, we can move the crate to a new workspace.

Reply:
https://github.com/aws/s2n-quic/pull/1433/commits/833e492c47954eb943e67afa1e14f70ef3d85d86